src/seq.rs

```diff
@@ -1359,7 +1359,7 @@ impl MidiEvent {
     }
 
     /// In case of success, returns a tuple of (bytes consumed from buf, found Event).
-    pub fn encode<'a>(&'a mut self, buf: &[u8]) -> Result<(usize, Option<Event<'a>>)> {
+    pub fn encode(&mut self, buf: &[u8]) -> Result<(usize, Option<Event>)> {
         // The ALSA documentation clearly states that the event will be valid as long as the Encoder
```

Comment:
Because of the unsafe trickery going on inside this method, I'd like to keep this one as it is to make it clear what's going on.

Reply:
``pub fn encode(&'_ mut self, buf: &[u8]) -> Result<(usize, Option<Event<'_>>)> {``
Could it work that way?

Reply:
That's still less clear than the original. Don't you agree?

Reply:
Yeah.